pkg/cloud/aws/actuators/cluster/actuator.go

```diff
@@ -64,14 +64,48 @@ func (a *Actuator) Reconcile(cluster *clusterv1.Cluster) error {
 	elbsvc := elb.NewService(scope)
 
 	// Store some config parameters in the status.
-	if len(scope.ClusterConfig.CACertificate) == 0 {
-		caCert, caKey, err := certificates.NewCertificateAuthority()
+	if !scope.ClusterConfig.CAKeyPair.HasCertAndKey() {
+		caCert, caKey, err := actuators.GetOrGenerateKeyPair(&scope.ClusterConfig.CAKeyPair, "cluster-ca")
```

Comment:
it might be worth extracting this whole thing into a certificates service so it follows the elb/ec2 pattern, but that can be an add-on, just make a ticket and assign to me please

Reply:
yeah, I agree. That'll be cleaner. I'll create an issue for that.

Reply:
created https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/537